kong/plugins/oauth2/crypto.lua

```diff
@@ -0,0 +1,28 @@
+-- Module to hash the basic-auth credentials password field
+local sha256 = require "resty.sha256"
+local to_hex = require "resty.string".to_hex
+local resty_random = require "resty.random"
+local assert = assert
+
+
+local function hash_secret_with_salt(client_secret, salt)
+  if client_secret == nil or client_secret == ngx.null then
+    client_secret = ""
+  end
+
+  local salted = client_secret .. salt
+  local digest = sha256:new()
+  assert(digest:update(salted))
+  return to_hex(digest:final())
+end
+
+return {
+  hash = function(secret)
```

Comment:
We should use peer-reviewed hashing algorithms like bcrypt instead of rolling our own here, which automatically handle things like salts and performance tunables